### PR TITLE
bs4 implementation 

### PR DESCRIPTION
**Pull request by blacksmithop**
## Change Overview:
Uses `bs4` to scrape requirements mentioned in [issue](https://github.com/Nysa-clan/DeepLI/issues/8)
Current format:
image
`question`, `url` and `source_verified` was acquired relatively easily. `answer` is the combination of commands being mentioned in the accepted answer. Due to the nature of the filter being applied (ranked by votes) only accepted answer is taken as the `answer`. `manual_verified` and `final_answer` defaults to False and None since they need to be verified manually.

## Testing Overview:
Creating a `dict` for each `url` and appending to a `list` is memory intensive and takes a while to finish. 
Can be replaced with writing to csv to remove that overhead.

## Related Issues / PRs:
Further testing will have to be done remotely with larger `PAGE_MAX`.

### CODE REVIEW

**Review by navan0 — APPROVED** [review]

**Comment by navan0** [comment]
@GopikrishnanSasikumar check and merge

File: dataset/datasetGen.py [hunk 1]
```diff
@@ -9,31 +10,52 @@ def Tag_To_Url(base: str, tag: str, page: int):
...
     tag_url = findall(r'<h3><a href="(.*?)" class', str(content))[1:]
     tag_url = [f'{base}{link}' for link in tag_url]
-    #return list of urls for that tag
+    # return list of urls for that tag
     yield tag_url
 
 
-def Link_To_JSON(urls: list):
-    url = urls[0]
+def Url_To_Data(url: str):
```
Comment (gksoriginals) [inline dataset/datasetGen.py]: Wouldn't it be great if we can split `url_to_data()` into small methods. This violate [SLAP](http://principles-wiki.net/principles:single_level_of_abstraction)

File: dataset/datasetGen.py [hunk 2]
```diff
@@ -9,31 +10,52 @@ def Tag_To_Url(base: str, tag: str, page: int):
...
-    is_verified = findall(r'"answer accepted-answer"', str(content))
-    if is_verified:
-        is_verified = True
-    else:
-        is_verified = False
-    result = {
-        'title': title,
-        'code': code,
-        'verfied': is_verified
+    soup = BeautifulSoup(str(content), 'html.parser')
```
Comment (GopikrishnanSasikumar) [inline dataset/datasetGen.py:21]: I would initialise `BeautifulSoup` outside the method.

**Review by gksoriginals — CHANGES_REQUESTED** [review]

File: dataset/datasetGen.py (hunk 1 above)
Comment (blacksmithop) [reply dataset/datasetGen.py]: Actually, that sounds better than having all the operations be done under one hood. I'll look into what can be moved around.

**Comment by blacksmithop** [comment]
image
Rewrote scraping logic to fit a class structure.
`DataSetGen` now has class variables `PAGE_MAX` and `platform_to_tag` . Use of class vars to share current data among methods.
All operations including collection of urls, and the operations under `url_to_data` are now class methods.
Namely `get_question`, `is_verified`, (returns a Boolean as well the url if applicable), `get_answer`.
This was run on a remote Linux environment with a `PAGE_MAX` of 1 and one tag for askubuntu. (screenshot)
Only accepted answers are considered by `data_from_url` (responsible for building the collection of data items)

**Review by navan0 — APPROVED** [review]
Looks good

**Pull request merged**